src/routes/safe/components/Apps/components/AppFrame.tsx

```diff
@@ -81,6 +81,10 @@ const INITIAL_CONFIRM_TX_MODAL_STATE: ConfirmTransactionModalState = {
   params: undefined,
 }
 
+const safeAppWeb3Provider = new Web3.providers.HttpProvider(getSafeAppsRpcServiceUrl(), {
```

Comment:
@mikheevm Would it be useful to create this provider in `getWeb3` file? This way we will only create it once and we can reuse it for every safe app that we open, creating less garbage and making it easier in case we need to reuse it on any other sections in future.

I would create it on `getWeb3` like we create `web3ReadOnly`

Reply:
Why if it's meant to be used only in this file? I don't get it

Reply:
We also don't want this RPC to be used in any other place than Safe Apps

Reply:
We can keep it like this for now. I have to deeply check garbage creation when creating and destroying this component. I'm not sure if the app will reuse same Web3 instance each time we open an app

Reply:
I'm not sure how moving it to `getWeb3` file solves the garbage collection problem because we'd import a reference to the object and it'd still live in memory because it's not needed. I think we can also move this variable to the component body so as soon as the component is destroyed the object is released from memory too